spec/API_specification/data_type_functions.md

```diff
@@ -7,6 +7,68 @@ A conforming implementation of the array API standard must provide and support t
 <!-- NOTE: please keep the constants in alphabetical order -->
 
 ## Objects in API
+(broadcast-arrays)=
+### broadcast_arrays(args, /, *)
+
+Broadcast any number of arrays against each other.
+
+#### Parameters
+
+-   **args**: [_&lt;array&gt;_]
+
+    -   arrays to broadcast.
+
+#### Returns
+
+-   **out**: [_&lt;array&gt;_]
+
+    -   readonly views on the original arrays. Furthermore, more than one element of a broadcasted array may refer to a single memory location.
+
+
+(broadcast-to)=
+### broadcast_to(array, /, shape, /)
+
+Broadcast an array to a new shape.
+
+-   A `ValueError` exception must be raised if the array is not compatible with the new shape according  to the semantics defined in {ref}`broadcasting`.
+
+#### Parameters
+
+-   **array**: _&lt;array&gt;_
+
+    -   array to broadcast.
+
+-   **shape**: _&lt;dtype&gt_
+
+    -   shape of the desired array.
+
+#### Returns
+
+-   **out**: _&lt;array&gt;_
+
+    -   a readonly view on the original array with the given shape where more than one element of a broadcasted array may refer to a single memory location.
+
+(can-cast)=
+### can_cast(from_, /, to, /)
+
+Determines if a type conversion is allowed under the rules described by {ref}`type-promotion`.
+
+#### Parameters
+
+-   **from_**: _Union\[ &lt;dtype&gt ]_
+
+    -   data type, scalar or array to cast from.
```

Comment:
This needs to be discussed - numpy allows Python scalars and arrays here. Other libraries may not. Would be nice to see the data on which libraries do what before making the decision.

Reply:
Given the docs numpy, jax and cupy seem to support scalars, arrays and dtypes. I'm going to add arrays for now and I'll check the usage of this function

Reply:
Edit: reviewing the python-record-api repo it seems that scalars and arrays are unused in the libraries we ran tests for. They are just using dtypes, should we just leave dtypes in the spec?

Reply:
Thanks for digging up the data here. I agree, let's just leave `dtype`.

Reply:
I think both dtypes and arrays should be supported here. This would be consistent with the `result_type` API which also supports both dtypes and arrays.